data-prepper-plugins/blocking-buffer/src/main/java/org/opensearch/dataprepper/plugins/buffer/blockingbuffer/BlockingBuffer.java

```diff
@@ -147,21 +147,16 @@ public void doWriteAll(Collection<T> records, int timeoutInMillis) throws Except
     @Override
     public Map.Entry<Collection<T>, CheckpointState> doRead(int timeoutInMillis) {
         final List<T> records = new ArrayList<>();
-        final Stopwatch stopwatch = Stopwatch.createStarted();
-        try {
+
+        if (timeoutInMillis == 0) {
+            blockingQueue.drainTo(records, batchSize);
+        } else {
+            final Stopwatch stopwatch = Stopwatch.createStarted();
             while (stopwatch.elapsed(TimeUnit.MILLISECONDS) < timeoutInMillis && records.size() < batchSize) {
-                final T record = blockingQueue.poll(timeoutInMillis, TimeUnit.MILLISECONDS);
-                if (record != null) { //record can be null, avoiding adding nulls
```

Comment:
This is null checking a concern?

Reply:
We do this null check on `poll` but not on `drainTo`. 
Poll only return one record and we add it to records Collection if it's not null. Removing this doesn't effect anything in my opinion because we add null records in `drainTo` call

Reply:
+1 to what Asif said

Reply:
Good point about that.

We should probably not have null records on write, but that is another matter.